provisioning_templates/PXELinux/rancheros_pxelinux.erb

```diff
@@ -0,0 +1,12 @@
+<%#
+kind: PXELinux
+name: RancherOS PXELinux
+model: ProvisioningTemplate
+oses:
+- RancherOS
+-%>
+DEFAULT rancheros
+
+LABEL rancheros
+  KERNEL <%= @kernel -%> rancher.password='changeme' rancher.state.dev=LABEL=RANCHER_STATE rancher.state.autoformat=[/dev/xvda] rancher.cloud_init.datasources=['url:<%= foreman_url('provision')-%>']
+  APPEND initrd=<%= @initrd -%>
```

Comment:
Foreman can provide root password via `<%= root_pass %>` but it will be in Linux crypt form (`$5$xxxx$yyyyyyy`), not sure if RancherOS accepts it in this form. Putting password via plaintext is something you do not want to do, this file is word-readable via TFTP! I suggest to delete whole `rancher.password='changeme'` part if the OS boots without it.

Device should be parametrized to make this template usable, your will only work for Xen, use this: `<%= host_param('autoformat_device') %>` or similar.

Reply:
Rancher don't support crypted passwor I tried. So I will delete this parameter. I will be the responsability of the user to either have public keys confgured or add this parameter.

Reply:
Thanks, since it supports cloud-init this is not an issue actually.

Reply:
The line is a bit off now, can you add two leading spaces?

This would work as well: `<%= host_param('autoformat_device') || '/dev/vda' %>

Reply:
The `install-disk` parameter should be re-used here, IMHO